### PR TITLE
feat: video lead asset (web)

### PR DESCRIPTION
**Pull request by BernieSumption**
Support video lead assets for articles. Doing this required:

1. Fixing the layout of the BrightcoveVideo component
2. Refactoring the article.web.js layout to use pure css to move and resize the lead asset at different breakpoints, instead of duplicating the lead asset elements and showing/hiding them on different breakpoints.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7b83598ef7190cb019533cfdb7e1ea5e03f8fde5/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/610e5211c05c548c7d40a2692e4f9b16931622b5/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/cfb028bc8a03f90d8a189ca651cbaa9da6ac614d/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/c8520f0530cb4cb29d2c5182b9e2e62279afaae0/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/02eb936eb71b91b817cce4aaa00fffba21081c7f/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/43ded5026b6f5778ddece65fdc72404bb4f4bbba/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/8a632ef97693b237e5d513b653d7a824a4bbe2df/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/18d779cdfc2cfd5821d3bb635f07a45b3926c903/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/20a687d817c414cc0f649b83cb8af2794ded5a0b/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/36ad01e9f18a69f140ed9c802ee3d438e69312ca/index.html

File: packages/article/styles/article-body/responsive.web.js [hunk 1]
```diff
@@ -10,14 +10,20 @@ import config from "../responsive-config";
...
-  wideUp: () => `width: ${config.wideBpWidth};`
-});
+export const ParagraphContainer = withResponsiveStyles(
+  "div",
+  {
+    base: () => config.articleContainerPadding,
+    mediumUp: () => config.mediumBpPositioning,
+    wideUp: () => `width: ${config.wideBpWidth};`
+  },
+  "ParagraphContainer"
```
Comment (nehasri89) [inline packages/article/styles/article-body/responsive.web.js]: After this PR is merged [2220](https://github.com/newsuk/cps-content-render/pull/2220), it uses babel-plugin-styled-components, which would do the trick we are trying to do here

**Comment by nehasri89** [comment]
I think this would be a candidate for breaking change? as we have added the styles on div that are hacked on the render in a different way

File: packages/article/styles/article-body/responsive.web.js (hunk 1 above)
Comment (killanaca) [reply packages/article/styles/article-body/responsive.web.js]: `add-react-displayname` is in master already

File: packages/responsive-styles/responsive-styles.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ export const config = {
 
 // Turn off prettier for this as the identified changes break this component
 // prettier-ignore
-const withResponsiveStyles = (Component, styles = {}) => styled(Component)`
+const withResponsiveStyles = (Component, styles = {}, displayName = null) => {
```
Comment (nikhedonia) [inline packages/responsive-styles/responsive-styles.js]: Components have already a displayName, extract it from there

File: packages/article/styles/article-body/responsive.web.js (hunk 1 above)
Comment (nikhedonia) [reply packages/article/styles/article-body/responsive.web.js]: `add-react-displayname` would not fix it as it handles slightly different cases. eg:
case 1:

```js
const MyTitle = () => <h1>Title </h1>;
MyTitle.displayName = "MyTitle"; //injected by plugin 
<MyTitle/>
```
```js
class MyTitle extends Component {
//...
};
MyTitle.displayName = "MyTitle"; //injected by plugin 
<MyTitle/>
```

Comment (nehasri89) [reply packages/article/styles/article-body/responsive.web.js]: 👍

File: packages/responsive-styles/responsive-styles.js (hunk 2 above)
Comment (BernieSumption) [reply packages/responsive-styles/responsive-styles.js]: @nikhedonia The issue here is that in the majority of cases, `withResponsiveStyles` is being used to create a styled container from a generic `View`, and so without a custom name supplied every component in the tree is called "Styled(View)". The correct name for the generated component is the same as the variable that the new component is assigned to. To pick a real example:

```js
const Heading = withResponsiveStyles(Text, {
  base: () => `
    font-family: ${fonts.headline};
    font-size: ${fontSizes.leadHeadline}px;
    text-align: center;
    color: ${colours.functional.brandColour};
    margin-bottom: 12px;
  `
});
```

The component should be called "Heading". AFAIK there's currently no way to do this automatically, short of writing our own babel transform or modifying babel-plugin-add-react-displayname.

Comment (BernieSumption) [reply packages/responsive-styles/responsive-styles.js]: Bear witness:

image

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/a635f07aedb926444f7a61e9c3f8df33fbdb56e5/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7332d4a986a96a5afe901e86d227278f98199845/index.html

File: packages/responsive-styles/responsive-styles.js (hunk 2 above)
Comment (nikhedonia) [reply packages/responsive-styles/responsive-styles.js]: makes sense

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/bef1b33f91146412af068b083a3f27124a27e0e8/index.html

**Pull request merged**